hw4/02-sessions.js

```diff
@@ -1,15 +1,102 @@
-const express = require('express');
-const session = require('express-session');
+const express = require("express");
+const session = require("express-session");
+const memstore = require("memorystore")(session); //Recommended by express-session docs
 const app = express();
 const port = process.env.PORT || 5000;
+const currentRoute = `Currently on route:  `;
+const routes = ["spongebob", "patrick", "main", "sandy"];
+let getRoutes = () => {
+  //Build a list of links for the root index.
+  let result = "";
+  routes.forEach(
+    (elem) => (result += `<li><a href="/${elem}">${elem}</a></li>`)
+  );
+  return result;
+};
+let routeResults = getRoutes();
 
-// Add your code here
+let visited = (history) => {
+  //Build an html list from the cookie path and return it.
+  let list = ``;
+  let array = history.split(";");
+  array.forEach((elem) => (list += `<li><a href="/${elem}">${elem}</a></li>`));
+  return list;
+};
 
-// Use the express-session module
-app.use(/** ... */);
+app.use(
+  session({
+    //Setting up the Express session.
+    cookie: { maxAge: 14_400 },
+    store: new memstore({
+      checkPeriod: 14_400,
+    }),
+    secure: false,
+    resave: true,
+    secret: "the fnords",
+  })
+);
 
-app.get('/', (req, res) => {
-  // Add your code here
+app.get("/", (req, res) => {
+  // /main redirects to root.
+  res.writeHead(200, { "Content-Type": "text/html" });
+  res.write(`<h2>Welcome to localhost:5000!</h2>`);
+  res.write(`<h3>${currentRoute}/</h3>`);
+  res.write(`Available Routes: `);
+  res.write(`<ul> ${routeResults} </ul>`);
```

Comment:
There are no fixed routes, I think in hw4 we can visit any routes of our choice.